### DISCUSSION TITLE
How to add it to the ESPHome to automatically receive updates?

### DISCUSSION THREAD

**Discussion by can-I-do** [gcormier/megadesk#144] [General]
Hi everyone, 

maybe this is a very stupid question, but I still cannot figure it out. Inside my ESPHome, I added other stuff there as well, I have the Logs and Update functionality for my ESP stuff 
![CleanShot 2023-10-17 at 10 40 31](https://github.com/gcormier/megadesk/assets/60129163/b0f87bc2-0eb7-4532-a6b9-d9d26ac61c69)

Now I added the Megadesk companion and it is working perfectly (and it is so awesome, thx for that!!), but it does not show up inside ESPHome, just under my devices: 
![CleanShot 2023-10-17 at 10 42 18](https://github.com/gcormier/megadesk/assets/60129163/4fef901b-63cf-43a6-aa35-1dfe4a3a8a3c)

How can I add it, so I can see logs and automatically receive updates ?

**Comment by gcormier** [comment]
Hello!

You will need to place `megadesk.h` in the appropriate folder in Home Assistant. This will depend how you have setup your HA, and you are best to check on the esphome/HA forum or discord. In my case I am using a docker installation method for Home Assistant, including esphome.

```
  esphome:
    container_name: esphome
    image: esphome/esphome
    environment:
      - TZ=America/Toronto
    volumes:
      - /docker/esphome:/config
```

With my setup, this means I copy `megadesk.h` to my mount point which is at `/docker/esphome` on my server.

![image](https://github.com/gcormier/megadesk/assets/8388983/0bec3347-5579-48ef-9ad5-314dfe704a12)


You can then add a new devices to the esphome control panel, skip the first time "flash", and then copy/paste your existing YAML configuration file. If you save and validate, it will be able to upload it as long as it can resolve the hostname properly.

**Reply by can-I-do** [reply]
Thx a lot for the help I will try my best to make it work :D

**Reply by can-I-do** [reply]
worked perfectly, thx!

edit: oh, but now it is not working any more :D 

edit2: got it to work again, but geting 

12:27:07 | [E] | [uart:015] | Reading from UART timed out at byte 0!
-- | -- | -- | --
12:27:07 | [E] | [uart:015] | Reading from UART timed out at byte 0!

as an error message, is that expected? I am using your standard megadesk_companion.yaml, is that correct?
